### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by DaniilPoiarkov**

### CODE REVIEW

File: backend/Tasque.BLL/MappingProfiles/TaskTemplateProfile.cs [hunk 1]
```diff
@@ -25,11 +25,11 @@ public TaskTemplateProfile()
 
             CreateMap<CosmosTemplateCustomField, TemplateCustomField>()
                 .ForMember(tt => tt.Dropdown, opt => opt
-                    .MapFrom(ct => ct.Type == TaskFieldType.Dropown ? JsonConvert.DeserializeObject<DropdownField>(ct.Content) : null))
+                    .MapFrom(ct => ct.Type == TaskFieldType.Dropown ? JsonConvert.DeserializeObject<DropdownField>(ct.Content ?? "") : null))
```
Comment (Its-OP) [inline backend/Tasque.BLL/MappingProfiles/TaskTemplateProfile.cs]: Change `""` to `string.Empty` here and at other places in the PR

File: backend/Tasque.BLL/Services/TaskService.cs [hunk 2]
```diff
@@ -115,8 +114,8 @@ public async Task<List<TaskDto>> GetAllProjectTasks(int projectId)
             return tasks.Join(customFields, t => t.Id, ca => int.Parse(ca.Id), (t, ca) =>
                 JoinTaskAttributesWithDto(t,
                     RenameFieldsWithActualValue(
-                        GetTaskTemplate(t.ProjectId, t.TypeId).Result,
-                            MapCosmosTaskFieldsToTaskCustomFields(t, ca.CustomFields).Result))).ToList();
+                        GetTaskTemplate(t.ProjectId, t.TypeId).Result?? new(),
```
Comment (Its-OP) [inline backend/Tasque.BLL/Services/TaskService.cs]: Add spacing before `??`

File: backend/Tasque.BLL/Services/TaskService.cs [hunk 3]
```diff
@@ -235,13 +247,21 @@ public async Task<List<TaskCustomFields>> MapCosmosTaskFieldsToTaskCustomFields(
             var template = await GetTaskTemplate(task.ProjectId, task.TypeId);
             var result = new List<TaskCustomFields>();
 
-            fields.ForEach(f => result.Add(new()
+            if (fields == null || fields.Count() == 0 || template == null || template.CustomFields == null)
```
Comment (Its-OP) [inline backend/Tasque.BLL/Services/TaskService.cs]: Use `template?.CustomFields == null`

File: backend/Tasque.BLL/Services/TaskService.cs [hunk 4]
```diff
@@ -235,13 +247,21 @@ public async Task<List<TaskCustomFields>> MapCosmosTaskFieldsToTaskCustomFields(
...
-                FieldType = template.CustomFields.Find(t => t.FieldId == f?.FieldId).Type,
-                FieldValue = f.FieldValue,
-            }));
+                var type = template.CustomFields.Find(t => t.FieldId == f?.FieldId)?.Type;
+
+                result.Add(new()
+                {
+                    FieldId = f.FieldId,
+                    FieldName = template?.CustomFields?.Find(t => t.FieldId == f?.FieldId)?.Name,
+                    FieldType = type?? TaskFieldType.Text,
```
Comment (Its-OP) [inline backend/Tasque.BLL/Services/TaskService.cs]: spacing

File: backend/Tasque.BLL/Services/TaskService.cs [hunk 5]
```diff
@@ -115,8 +114,8 @@ public async Task<List<TaskDto>> GetAllProjectTasks(int projectId)
             return tasks.Join(customFields, t => t.Id, ca => int.Parse(ca.Id), (t, ca) =>
                 JoinTaskAttributesWithDto(t,
                     RenameFieldsWithActualValue(
-                        GetTaskTemplate(t.ProjectId, t.TypeId).Result,
-                            MapCosmosTaskFieldsToTaskCustomFields(t, ca.CustomFields).Result))).ToList();
+                        GetTaskTemplate(t.ProjectId, t.TypeId).Result?? new(),
+                            MapCosmosTaskFieldsToTaskCustomFields(t, ca.CustomFields?? new()).Result)?? new())).ToList();
```
Comment (Its-OP) [inline backend/Tasque.BLL/Services/TaskService.cs]: spacing

File: backend/Tasque.BLL/Services/TaskStateService.cs [hunk 6]
```diff
@@ -16,7 +16,7 @@ public TaskStateService(DataContext db, IMapper mapper, CurrentUserParameters cu
         
         }
 
-        public new async Task<IEnumerable<TaskState>> GetAll()
+        public async Task<IEnumerable<TaskState>> GetAll()
         {
             var tasksStates = await _db.TaskStates
```
Comment (Its-OP) [inline backend/Tasque.BLL/Services/TaskStateService.cs]: Remove `async` from the method's signature and just do `return _db.TaskStates.ToListAsync()`

File: backend/Tasque.BLL/Services/TaskTypeService.cs [hunk 7]
```diff
@@ -20,7 +20,7 @@ public List<TaskTypeDto> GetAllTaskTypesByProjectId(int projectId)
             return _mapper.Map<List<TaskTypeDto>>(_db.TaskTypes.Where(t => t.ProjectId == projectId));
         }
 
-        public new async Task<IEnumerable<TaskType>> GetAll()
+        public async Task<IEnumerable<TaskType>> GetAll()
```
Comment (Its-OP) [inline backend/Tasque.BLL/Services/TaskTypeService.cs]: Same as above

File: backend/Tasque.Core.Common/Entities/User.cs [hunk 8]
```diff
@@ -15,6 +15,7 @@ public User()
         OwnedTasks = new List<Task>();
         ParticipatedTasks = new List<Task>();
         Roles = new List<UserProjectRole>();
+        SystemRoles = new List<UserOrganizationRole>();
```
Comment (Its-OP) [inline backend/Tasque.Core.Common/Entities/User.cs:18]: I don't think that `SystemRole` is a good title here

File: frontend/src/app/components/tasque-board/tasque-board.component.html [hunk 9]
```diff
@@ -1,4 +1,11 @@
-<div class="board" *ngIf="isShow && project !== undefined && columns !== undefined">
+<div *ngIf="!isShow">
+  <mat-spinner 
+    style="margin: 250px auto;" 
+    mode="indeterminate"
+    >
+  </mat-spinner>
+</div>
+<div class="board" *ngIf="isShow && project !== undefined && columns !== undefined && projectTasks != undefined">
```
Comment (Its-OP) [inline frontend/src/app/components/tasque-board/tasque-board.component.html]: `isShow && project && columns && projectTasks`

File: backend/Tasque.BLL/Services/TaskService.cs (hunk 3 above)
Comment (DaniilPoiarkov) [reply backend/Tasque.BLL/Services/TaskService.cs]: It's nnecessary coz previous condition checks if they're null

File: backend/Tasque.Core.Common/Entities/User.cs (hunk 8 above)
Comment (DaniilPoiarkov) [reply backend/Tasque.Core.Common/Entities/User.cs:18]: It's already in dev, just removing backend warning

**Review by FeloniukM — DISMISSED** [review]

File: backend/Tasque.BLL/Services/TaskService.cs (hunk 3 above)
Comment (Its-OP) [reply backend/Tasque.BLL/Services/TaskService.cs]: `template?.CustomFields == null` will return `null` when `teamplate` is `null` due to the `?` operator

**Review by Its-OP — APPROVED** [review]

**Review by FeloniukM — APPROVED** [review]

**Pull request merged**